src/components/Player/Pages/Matches/Matches.jsx

```diff
@@ -23,47 +24,47 @@ Matches.propTypes = {
   strings: PropTypes.shape({}),
 };
 
-const getData = (props) => {
-  props.getPlayerMatches(props.playerId, props.location.search);
-};
-
-class RequestLayer extends React.Component {
-  static propTypes = {
-    location: PropTypes.shape({
-      key: PropTypes.string,
-    }),
-    playerId: PropTypes.string,
-    strings: PropTypes.shape({}),
-  }
-
-  componentDidMount() {
-    getData(this.props);
+const getData = (props, gamemode) => {
+  let params = props.location.search;
+  if (gamemode === 'turbo') {
+    params = `${props.location.search}&game_mode=23`;
   }
+  props.getPlayerMatches(props.playerId, params);
+};
 
-  componentDidUpdate(prevProps) {
-    if (this.props.playerId !== prevProps.playerId || this.props.location.key !== prevProps.location.key) {
-      getData(this.props);
+const RequestLayer = (props) => {
+  const gamemodeState = useGamemode();
```

Comment:
don't think this is necessary, if you move the logic to the actions file and just check localstorage there. You may not be able to use Context there as that's not in the React tree.